diff --git a/drivers/irqchip/irq-imx-irqsteer.c b/drivers/irqchip/irq-imx-irqsteer.c
index a5165579b..c62b720b7 100644
--- a/drivers/irqchip/irq-imx-irqsteer.c
+++ b/drivers/irqchip/irq-imx-irqsteer.c
@@ -191,7 +191,7 @@ static int imx_irqsteer_chans_enable(struct irqsteer_data *data)
 {
 	int ret;
 
-	ret = clk_prepare_enable(irqsteer_data->ipg_clk);
+	ret = clk_prepare_enable(data->ipg_clk);
 	if (ret) {
 		dev_err(data->dev, "failed to enable ipg clk: %d\n", ret);
 		return ret;
@@ -378,7 +378,6 @@ static int imx_irqsteer_runtime_resume(struct device *dev)
 
 	return 0;
 }
-#endif
 
 static const struct dev_pm_ops imx_irqsteer_pm_ops = {
 	SET_NOIRQ_SYSTEM_SLEEP_PM_OPS(pm_runtime_force_suspend,
@@ -386,6 +385,7 @@ static const struct dev_pm_ops imx_irqsteer_pm_ops = {
 	SET_RUNTIME_PM_OPS(imx_irqsteer_runtime_suspend,
 			   imx_irqsteer_runtime_resume, NULL)
 };
+#endif
 
 static const struct of_device_id imx_irqsteer_dt_ids[] = {
 	{ .compatible = "fsl,imx-irqsteer", },
@@ -397,7 +397,9 @@ static struct platform_driver imx_irqsteer_driver = {
 	.driver = {
 		.name = "imx-irqsteer",
 		.of_match_table = imx_irqsteer_dt_ids,
+#ifdef CONFIG_PM_SLEEP
 		.pm = &imx_irqsteer_pm_ops,
+#endif
 	},
 	.probe = imx_irqsteer_probe,
 	.remove = imx_irqsteer_remove,
