--- a/arch/arm/lib/bitops.h
+++ b/arch/arm/lib/bitops.h
@@ -12,7 +12,7 @@
 	and	r3, r0, #31		@ Get bit offset
 	mov	r0, r0, lsr #5
 	add	r1, r1, r0, lsl #2	@ Get word offset
-#if __LINUX_ARM_ARCH__ >= 7 && defined(CONFIG_SMP)
+#if __LINUX_ARM_ARCH__ >= 7 && defined(CONFIG_SMP) && !defined(GENODE_PREVENT_PLDW)
 	.arch_extension	mp
 	ALT_SMP(W(pldw)	[r1])
 	ALT_UP(W(nop))
@@ -39,7 +39,7 @@
 	add	r1, r1, r0, lsl #2	@ Get word offset
 	mov	r3, r2, lsl r3		@ create mask
 	smp_dmb
-#if __LINUX_ARM_ARCH__ >= 7 && defined(CONFIG_SMP)
+#if __LINUX_ARM_ARCH__ >= 7 && defined(CONFIG_SMP) && !defined(GENODE_PREVENT_PLDW)
 	.arch_extension	mp
 	ALT_SMP(W(pldw)	[r1])
 	ALT_UP(W(nop))
--- a/arch/arm/include/asm/processor.h
+++ b/arch/arm/include/asm/processor.h
@@ -114,7 +114,7 @@
 		:: "p" (ptr));
 }
 
-#if __LINUX_ARM_ARCH__ >= 7 && defined(CONFIG_SMP)
+#if __LINUX_ARM_ARCH__ >= 7 && defined(CONFIG_SMP) && !defined(GENODE_PREVENT_PLDW)
 #define ARCH_HAS_PREFETCHW
 static inline void prefetchw(const void *ptr)
 {
