diff --git a/board/boundary/nitrogen8m_som/nitrogen8m_som.c b/board/boundary/nitrogen8m_som/nitrogen8m_som.c
index 89ebf8e08b..3269cdccee 100644
--- a/board/boundary/nitrogen8m_som/nitrogen8m_som.c
+++ b/board/boundary/nitrogen8m_som/nitrogen8m_som.c
@@ -335,6 +335,10 @@ static void set_env_vars(void)
 	env_set("fdt_addr", "0x50000000");
 #ifdef MNTREFORM_BOOT_RISCOS
 	env_set("bootcmd", "usb start; usb reset; ext4load mmc 1 ${loadaddr} /RISCOSIMX8M; go ${loadaddr} 32");
+#elif MNTREFORM_BOOT_GENODE
+	env_set("bootcmd", "ext4load mmc 1 0x51000000 /uImage; bootm start 0x51000000; bootm loados; bootm go");
+#elif MNTREFORM_BOOT_GENODE_FROM_EMMC
+	env_set("bootcmd", "ext4load mmc 0 0x51000000 /uImage; bootm start 0x51000000; bootm loados; bootm go");
 #elif MNTREFORM_BOOT_EMMC
 	// boot rescue system from eMMC
 	env_set("bootargs", "noinitrd root=/dev/mmcblk0p1 rootwait rw console=ttymxc0,115200 console=tty1 cma=512M pci=nomsi init=/sbin/reform-init");
diff --git a/common/bootm.c b/common/bootm.c
index e517d9f118..a349bbb6a3 100644
--- a/common/bootm.c
+++ b/common/bootm.c
@@ -31,7 +31,7 @@
 
 #ifndef CONFIG_SYS_BOOTM_LEN
 /* use 8MByte as default max gunzip size */
-#define CONFIG_SYS_BOOTM_LEN	0x800000
+#define CONFIG_SYS_BOOTM_LEN	0x4000000
 #endif
 
 #define IH_INITRD_ARCH IH_ARCH_DEFAULT
