--- linux/arch/arm64/boot/dts/freescale/imx8mq-evk.dts
+++ linux/arch/arm64/boot/dts/freescale/imx8mq-evk.dts
@@ -103,6 +103,89 @@
 		spdif-controller = <&spdif2>;
 		spdif-in;
 	};
+
+	soc@0 {
+		bus@32c00000 {
+
+			hdmi: hdmi@32c00000 {
+				status = "okay";
+
+				compatible = "cdn,imx8mq-hdmi";
+				reg = <0x32c00000 0x100000>,
+					<0x32e40000 0x40000>;
+				interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>,
+							 <GIC_SPI 25 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "plug_in", "plug_out";
+
+				lane-mapping = <0xe4>; // 0=0, 1=1, 2=2, 3=3 (no swapping)
+
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					port@0 {
+						reg = <0>;
+						#address-cells = <1>;
+						#size-cells = <0>;
+						hdmi_in: endpoint@0 {
+							reg = <0>;
+							remote-endpoint = <&dcss_hdmi_out>;
+						};
+					};
+				};
+			};
+
+			dcss: display-controller@32e00000 {
+				status = "okay";
+
+				compatible = "nxp,imx8mq-dcss";
+				reg = <0x32e00000 0x2d000>, <0x32e2f000 0x1000>;
+				interrupts = <6>, <8>, <9>;
+				interrupt-names = "ctxld", "ctxld_kick", "vblank";
+				interrupt-parent = <&irqsteer>;
+				clocks = <&clk IMX8MQ_CLK_DISP_APB_ROOT>,
+					<&clk IMX8MQ_CLK_DISP_AXI_ROOT>,
+					<&clk IMX8MQ_CLK_DISP_RTRM_ROOT>,
+					<&clk IMX8MQ_VIDEO2_PLL_OUT>,
+					<&clk IMX8MQ_CLK_DISP_DTRC>,
+					<&clk IMX8MQ_VIDEO2_PLL1_REF_SEL>,
+					<&clk IMX8MQ_CLK_PHY_27MHZ>;
+				clock-names = "apb", "axi", "rtrm", "pix", "dtrc", "pll_src", "pll_phy_ref";
+				assigned-clocks = <&clk IMX8MQ_CLK_DISP_AXI>,
+					<&clk IMX8MQ_CLK_DISP_RTRM>,
+					<&clk IMX8MQ_VIDEO2_PLL1_REF_SEL>,
+					<&clk IMX8MQ_VIDEO_PLL1_REF_SEL>,
+					<&clk IMX8MQ_CLK_DC_PIXEL>;
+				assigned-clock-parents = <&clk IMX8MQ_SYS1_PLL_800M>,
+					<&clk IMX8MQ_SYS1_PLL_800M>,
+					<&clk IMX8MQ_CLK_27M>,
+					<&clk IMX8MQ_CLK_25M>,
+					<&clk IMX8MQ_VIDEO_PLL1_OUT>;
+				assigned-clock-rates = <800000000>,
+					<400000000>,
+					<27000000>,
+					<25000000>,
+					<594000000>;
+
+				// external display (HDMI)
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					port@0 {
+						reg = <0>;
+						#address-cells = <1>;
+						#size-cells = <0>;
+						dcss_hdmi_out: endpoint@0 {
+							reg = <0>;
+							remote-endpoint = <&hdmi_in>;
+						};
+					};
+				};
+
+			};
+		};
+	};
 };
 
 &A53_0 {
